lib/Agrammon/Web/Routes.pm6

```diff
@@ -251,12 +251,42 @@ sub api-routes (Str $schema, $ws) {
                 }
             }
         }
+        operation 'getOutputVariables', -> LoggedIn $user {
+            request-body -> ( :datasetName($dataset-name)! ) {
+                my %output = $ws.get-output-variables($user, $dataset-name);
+                content 'application/json', { %output };
+                CATCH {
+                    note "$_";
+                    response.status = 500;
+                    content 'application/json', %( error => $_ )
+                }
+            }
+        }
+        operation 'getInputVariables', -> LoggedIn $user {
+            request-body -> ( :datasetName($dataset-name)! ) {
+                my %data = $ws.get-input-variables;
+                %data<datasetName> = $dataset-name;
+                # debug output on purpose
+                # dd "KEYS:", %data.keys;
+                # dd "GRAPHS:", %data<graphs>;
+                # dd "REPORTS:", %data<reports>;
+                # dd "INPUTS[0]:", %data<inputs>[0];
+                # for %data<inputs>[] -> $input {
+                #     dd $input<optionsLang>;
+                # }
```

Comment:
If it's going to stay around, consider putting it behind an AGRAMMON_DEBUG env var.

Reply:
I'll remember that suggestion, but removed that particular code as I hopefully won't need it anymore.